### PR TITLE
Sync human readable input with UI

### PR DESCRIPTION
**Pull request by mitogh**
Make sure UI and input are in sync so if any changes the other reacts to those changes in the UI or the input.

Ticket: https://central.tri.be/issues/112877

### CODE REVIEW

File: plugins/common/src/modules/utils/date.js [hunk 1]
```diff
@@ -51,25 +57,90 @@ export const timezones = () => {
...
+		if ( momentUtil.isSameDay( from.moment, to.moment ) ) {
+			/**
+			 * If both dates are happening on the same day only the time is relevant on the secnod
+			 * part of the string to keep string cleaner like: 'Oct 8 2018 at 12:00 pm - 12:30 pm'
+			 * instead of 'Oct 8 2018 at 12:00 pm - Oct 8 2018 at 12:30 pm'
+			 */
+			parts.push( to.detail.time );
+		} else if ( momentUtil.isSameMonth( from.moment, to.moment ) ) {
+			/**
+			 * If both dates are happening on the same month and not on the same day
```
[the remark this answers is not in the document]
Comment (paulmskim) [reply plugins/common/src/modules/utils/date.js]: If you read the entire comment, it doesn't connect. Maybe it's just the examples that are incorrect?

**Comment by mitogh** [comment]
Comments / questions, have been addressed. 

Thanks

File: plugins/events/src/modules/data/blocks/datetime/sagas.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+ * when the input is handling a new change() events
+ *
+ * @since TBD
+ *
+ * @returns {IterableIterator<*>}
+ */
+export function* onHumanReadableChange() {
+	// Wait in case there's a new change on the input
+	const WAIT_PERIOD_IN_MILLISECONDS = 700;
+	yield call( delay, WAIT_PERIOD_IN_MILLISECONDS );
```
Comment (nealfennimore) [inline plugins/events/src/modules/data/blocks/datetime/sagas.js]: Why not cut out the variable and just put in `700`? :)

File: plugins/events/src/modules/data/blocks/datetime/sagas.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+ * Generator used to reset the label of the input using the current values on the start and end date, useful when
+ * the user enters a data that is not valid or can't be parsed into a valid date so the input always reflects a valid
+ * date based on the current values set into the dates of the event.
+ *
+ * @since TBD
+ *
+ * @returns {IterableIterator<*>}
+ */
+export function* resetNaturalLanguageLabel() {
+	const dates = {
```
Comment (nealfennimore) [inline plugins/events/src/modules/data/blocks/datetime/sagas.js]: This doesn't need to change, but you could also do the same thing using [all](https://redux-saga.js.org/docs/api/#alleffects). Just an FYI

File: plugins/events/src/modules/data/blocks/datetime/sagas.js [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+ * @returns {IterableIterator<*>}
+ */
+export function* setHumanReadableLabel( dates = {} ) {
+	const currentLabel = yield select( selectors.getNaturalLanguageLabel );
+
+	if ( currentLabel === '' ) {
+		return;
+	}
+
+	const updatedLabel = rangeToNaturalLanguage( dates.start, dates.end );
```
Comment (nealfennimore) [inline plugins/events/src/modules/data/blocks/datetime/sagas.js]: Maybe convert this over to use `call`?

File: plugins/events/src/modules/data/blocks/datetime/sagas.js (hunk 2 above)
Comment (mitogh) [reply plugins/events/src/modules/data/blocks/datetime/sagas.js]: Sure, just wanted to increase the clarity on this one 👍

**Comment by mitogh** [comment]
Code has been updated @nealfennimore thanks 👍

**Pull request merged**